docs/general/builtins/aggregation.rst

```diff
@@ -169,6 +169,52 @@ array.
     +---------------------+
     SELECT 1 row in set (... sec)
 
+
+Example: Aggregate table names to their schema::
+
+
+    cr> select
+    ...    table_schema,
+    ...    array_agg(table_name) as tables
+    ... from information_schema.tables
+    ... group by table_schema;
+    +---------------------+-----------------------------------------+
+    | table_schema        |  tables                                 |
+    +---------------------+-----------------------------------------+
+    | doc                 | []                                      |
+    +---------------------+-----------------------------------------+
+    | sys                 | [cluster, job_metrics, privileges...]   |
+    +---------------------+-----------------------------------------+
+    | pg_catalog          | [pg_attribute, pg_roles, pg_proc...]    |
+    +---------------------+-----------------------------------------+
+    SELECT 3 row in set (... sec)
+
+
+Also accepts a dictionary of mapped columns::
+
+    cr> select
+    ...    table_schema,
+    ...    array_agg(
+    ...         {
+    ...           "table" = table_name,
+    ...           "replicas" = number_of_replicas,
+    ...           "shards" = number_of_shards
+    ...         }
+    ...     ) as tables
+    ... from information_schema.tables
+    ... group by table_schema;
```

Comment:
I think phrased this way it may give the impression that the dictionary handling is somehow special to `array_agg`, but that's not the case.

I'd rephrase this to clarify that.

```suggestion
``array_agg`` accepts all input types. This is often useful in combination with :ref:`object constructors <sql-object-constructor>`. An example::

    cr> select
    ...    table_schema,
    ...    array_agg(
    ...         {
    ...           "table" = table_name,
    ...           "replicas" = number_of_replicas,
    ...           "shards" = number_of_shards
    ...         }
    ...     ) as tables
    ... from information_schema.tables
    ... group by table_schema;
```


(sphinx ref is untested)

Reply:
I see, would it also make sense to change `column` from `array_agg(column)` to something else that also displays that it accepts more than just the column name?

Reply:
Maybe even have several definitions for the same function? much like postgres https://www.postgresql.org/docs/16/functions-aggregate.html

Reply:
> Maybe even have several definitions for the same function? much like postgres

I'm not really seeing the value in that for this case? The function always behaves the same and it has a single signature: `array_agg ( anyelement ) → anyelement[]`


> I see, would it also make sense to change column from array_agg(column) to something else that also displays that it accepts more than just the column name?

It doesn't accept more than one parameter. The point is that the expression can be any valid scalar expression. That includes object constructors, which in turn can use multiple other expressions. Same if you used another scalar as parameter that itself had more than 1 parameter..

That's kinda the point about my remark. There is nothing special about the `array_agg( { ...})` combination. I'm not opposed to including the example, but it should be clear that it's just using an object constructor.

And in general, the examples should each show a unique aspect of something. More != better. If you just repeat more of the same you get a worse signal/noise ratio.

Reply:
Sorry, I didn't mean 'more parameters' but 'different kind of parameters'.

I understand that more != better, but in this particular I didn't know it couldn't anything else than 'column' as the doc shows. 

In my case, I needed `array_agg({...})` but couldn't find any example or documentation at the time that'd lead me to use `any scalar expression`, so I gave up and thought that perhaps it had less functionality vs the postgress one, and I solved my issue with a query twice as complex and big, later on another iteration I asked in the community and I was shown a proper usage of it.

This PR is trying to solve a poor developer experience (from someone who recently started to use CrateDB to build stuff) that I had.

I agree with how you corrected and phrased the doc comment.